src/data_store/data_store_conduit.cpp

```diff
@@ -1172,7 +1172,7 @@ void data_store_conduit::allocate_shared_segment(std::unordered_map<int,size_t>
   }
   m_comm->trainer_barrier();
 
-  int shm_fd;
+  int shm_fd = -1;
```

Comment:
This is the important change. The rest are just whitespace changes because I have emacs configured to delete trailing whitespace before saving.